resolve_stream.py

```diff
@@ -93,6 +101,112 @@ def print_parameter(param, stream='', rd='', indent_level=0):
         parameter=param, stream=stream, rd=rd, indent=indent * indent_level)
 
 
+def print_qparameter(qparam, indent_level=0):
+    print_parameter(qparam.parameter, qparam.stream, qparam.refdes, indent_level=indent_level)
+
+
+class InvalidParameter(NameError):
+    """
+    Raised when parameter is invalid.
+
+    This occurs when attempting to resolve a parameter for a stream in which it does not belong.
+    """
+
+
+class UnresolvedParameter(UserWarning):
+    """
+    Indicates inability to resolve a parameter.
+
+    This occurs when the preload definition for a parameter is incorrect or when an expected stream
+    definition is missing.
+
+    Note - This does NOT necessary indicate an error. It is possible that a parameter may request
+    another parameter as an optional input.
+    """
+
+
+def resolve_parameter(refdes, method, stream, param, stream_map, m2m, depth=1):
+    """
+    Determine the primary set of parameters required to calculate a derived parameter.
+
+    This only fetches the initial set of parameters. If the initial set of parameters includes
+    functions, they will also need to be resolved with subsequent calls to this function.
+
+    :param refdes:  Fully qualified reference designator for instrument collecting the data.
+    :param method:  Manner in which data is collected (e.g. 'telemetered').
+    :param stream:  Stream name in which the parameter resides (e.g. ctdpf_sbe43_sample).
+    :param param:  Parameter to resolve (e.g. Parameter(seawater_temperature)).
+    :param stream_map:
```

Comment:
this is a lonely param, no documentation for it?

Reply:
I'll add documentation for `stream_map`